lib/crypto-tools/keyring.js

```diff
@@ -273,4 +281,95 @@ KeyRing.prototype._decrypt = function(data) {
   return JSON.parse(dec) && dec;
 };
 
+/**
+ * Retrieve the bucketKey
+ * @param {String} bucketId - bucket id
+ * @returns {String|null}
+ */
+KeyRing.prototype.generateBucketKey = function(bucketId) {
+  if(!this._mnemonic || !bucketId){
+    return null;
+  }
+  return DeterministicKeyIv.getDeterministicKey(this._mnemonic, bucketId);
```

Comment:
Or perhaps this is not needed, as it's just a hash of a normalized phrase, anyways.